### PR TITLE
Add Controls to Practice Driving

### LINKED ISSUE #42: Add controls to the practice driving screen
As a user, I want to be able to control the car so that I can practice my driving skills

**Acceptance Criteria:**
- [x] An intuitive way for steering the car is added
- [x] An intuitive way for accelerating/decelerating the car is added
- [x] The finger placement for the controls still allows for a dashboard to be added somewhere nicely
- [x] The controls are tested manually
- [x] The interface is understandable and can be accessed within at most 5 clicks.
- [x] pops up a window to ask “Are you sure you want to exit?” with “yes” or “no”

### PR DESCRIPTION
**Pull request by plepisnew**
Closes #42 

- Fixed acceleration and deceleration of previous UI controls: instead of updating instantaneous speed by using MQTT, simply store the current throttle percentage in a (observable) field and use that value.
- Added confirmation upon trying to exit driving screen.
- Joystick implementation left for Sprint 4

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java [hunk 1]
```diff
@@ -102,14 +110,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            }
-        }
+//        if (CarState.instance.isConnected()) {
+//            try {
+//                CarState.instance.getConnectedCar().changeSpeed(25.0);
+//            }
+//            catch(Exception ex) {
+//                // ignore
+//            }
+//        }
```
Comment (platisd) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java]: You shouldn't merge this while these lines of code are commented out. Either remove them altogether or fix the issue.
You may use it as it is for discussion and solving the issue of course.

**Comment by arrestoteles** [comment]
Alt text

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java (hunk 1 above)
Comment (plepisnew) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java]: Thanks for the feedback! Does this apply to all commented-out code?

Comment (Lancear) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java]: yes!

Comment (Lancear) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java:120]: Yes!

**Review by Lancear — DISMISSED** [review]
Noice, works greatly now!

**Review by JohnWebb00 — DISMISSED** [review]

**Review by JohnWebb00 — APPROVED** [review]

**Review by Lancear — APPROVED** [review]
image

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java (hunk 1 above)
Comment (platisd) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/PracticeDrivingActivity.java]: Yeap, unless there's a very significant reason there shouldn't be any commented out code committed to a main branch.
Some tools even detect and flag comments containing code as a "smell".

**Pull request merged**